### ISSUE TITLE
Writing UI test and making UI more stable

### ISSUE THREAD

**Issue by keshav234156** [publiclab/image-sequencer#1369]
With #1366 we were able to set up a basic browser-based UI test and it works flawlessly. So lets build more tests and make the UI stable.
Test yet to be built:

- [ ] Clicking add step adds a new box and a different image

- [ ] Clicking clear step button clear all the step

- [ ] Clicking Delete Step removes the image

- [ ] Select a module to have all the modules with the correct name(we faced issues when modules have which is undefined in the past ).

- [ ] Insert step works as expected.

- [ ] Input types(range,text etc) of all the modules are correct-
we need to fix this before we can write test on it

**Comment by keshav234156** [comment]
cc:-@jywarren @HarshKhandeparkar we can add more tests as we go along the way but these are most basic set of tests that are needed.

**Comment by jywarren** [comment]
This is FANTASTIC!!!! Thanks!!!

If we write the tests generically enough, using simple classnames, then the
new UI Harsh has been working on could be run against the same set of
tests! This would be amazing. Let's keep this in mind and Harsh, maybe try
to get your classnames to match so that your UI work passes these tests too?

**Comment by ataata107** [comment]
I am new here and would love to work on this.

**Comment by jywarren** [comment]
we'd love your help!!! Which would you like to try? Go ahead and open a
pull request with the new test; start with something simple and then keep
building on it!

**Comment by ataata107** [comment]
I will start with the basic ones like clear step button and remove step button and would start building from it. Thanks for the opportunity.

**Comment by SidharthBansal** [comment]
I have published this on gci dashboard

**Comment by SidharthBansal** [comment]
Thanks Keshav

**Comment by HarshKhandeparkar** [comment]
@jywarren sure. The new UI will just be less messy and it is coming together really well. I think I do not need to add the E2E tests which would require headless browsers. I will just add the unit tests with perhaps a virtual DOM. Is that ok? We can use the tests we are developing here, there.

We won't have to migrate anything, we can just copy the dist from there and use grunt here.

**Comment by pythongiant** [comment]
```
describe('Add step', () => {
  test('length is increased', async () => {
    	await page.waitForSelector('.step');
    const previousLength = await page.evaluate(() => document.querySelectorAll('.step').length);
    await page.click('[data-value=\'brightness\']');
    const previousLength1 = await page.evaluate(() => document.querySelectorAll('.step').length);
    expect(previousLength).toBe(1);
    expect(previousLength1).toBe(2);
  }, timeout);
});
```
Doesnt this check if a new step is added? @keshav234156 i am just a beginner sorry if i am missing something.

**Comment by pythongiant** [comment]
I would like to work on clear step test, Thanks

**Comment by keshav234156** [comment]
@pythongiant This piece of code tests that when the radio button is clicked a new step is added and length is increased . In the GCI task, you have test when we select a value from the dropdown and click ADD STEP a new is added and src of the previous step is different from src of the present step. @ataata107 is already working on the Clear step.

**Comment by pythongiant** [comment]
Thanks for the explaination @keshav234156 i must have skipped @ataata107 comment by mistake. Sorry bout that. Can i work on delete one?

**Comment by keshav234156** [comment]
@pythongiant It's easy.I will help you here .
with $("[data-value='blur']").click() You would be able to set dropdown value to blur
with $("#add-step-btn").click() you can click ADD STEP button

Now just that length has increased as done in above
and check src of both step is different

**Comment by Ryan10145** [comment]
I am currently working on the 4th one (Select a module dropdown having the correct names). I claimed the corresponding task on the GCI dashboard.

**Comment by keshav234156** [comment]
@pythongiant Are you still stuck? Feel free to ask any doubt related to the task.
@Ryan10145 great!!!

**Comment by pythongiant** [comment]
Umm @keshav234156 i was talking about working on the 3rd one. Is that not available atm?

**Comment by pythongiant** [comment]
![image](https://user-images.githubusercontent.com/13624560/71334609-de9cd980-2564-11ea-86ad-39bf16e28296.png)
@keshav234156 we are talking about this button right?

**Comment by keshav234156** [comment]
@pythongiant No,This is Add step button of insert Button.We are talking of insert button at the bottom.

**Comment by pythongiant** [comment]
![image](https://user-images.githubusercontent.com/13624560/71335705-06db0700-256a-11ea-86a8-b1cd45784c53.png)
this one then?

**Comment by keshav234156** [comment]
@pythongiant Yes

**Comment by Ryan10145** [comment]
Due to holiday celebrations and other commitments, I wasn't able to work on this for the last few days like I thought I would be able to. I also won't be able to work on it in the near future, so I am giving up the task and leaving it up for anyone to claim. Sorry for the unfortunate news.

**Comment by ataata107** [comment]
@pythongiant Are you working on the delete step?

**Comment by pythongiant** [comment]
Nope @ataata107

**Comment by ataata107** [comment]
Ok I will start working on it

**Comment by jywarren** [comment]
Hi all, this is coming along really well and I wanted to thank everyone for your help. This is doing a huge amount to stabilize our project!

**Comment by jywarren** [comment]
Noting Apply button test is ready in #1579, and maybe we need to update what still needs tests?

**Comment by jywarren** [comment]
One could be - a test of which steps get refreshed when we modify things. To ensure that only those tests that are "after" a change get refreshed; i think i've seen some unnecessary step reruns in manual testing.

Others we could test for?

**Comment by jywarren** [comment]
Noting a few that've been observed and we should test for in Jest:

* https://github.com/publiclab/image-sequencer/issues/1000#issuecomment-526683755
* https://github.com/publiclab/image-sequencer/issues/1238#issuecomment-527008195 - `Lol, I found a new bug. This is crazy. If you click on the step collapse btn and if you have multiple steps, clicking on the load-image step collapses something totally different. After some time, all the collapse btns break.`

Others we want to copy in here?

**Comment by jywarren** [comment]
Maybe we should check in on whether some of these tests have been written and could be checked off?
